File: vol1/java-examples/src/test/java/com/heatonresearch/aifh/error/TestErrorCalculationESS.java
package com.heatonresearch.aifh.error;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Test ESS errors.
 */
public class TestErrorCalculationESS {

    @Test
    public void testErrorCalc() {
        ErrorCalculation calc = new ErrorCalculationESS();
        double result = ErrorTestingUtil.calculateError(
                calc,
                ErrorTestingUtil.ACTUAL,
                ErrorTestingUtil.IDEAL);
        assertEquals(1516.205, result, 0.001);
    }
}

File: vol1/csharp-examples/UnitTests/Core/Error/TestErrorCalculationESS.cs
using AIFH_Vol1.Core.Error;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTests.Core.Error
{
    [TestClass]
    public class TestErrorCalculationESS
    {
        [TestMethod]
        public void TestErrorCalc()
        {
            IErrorCalculation calc = new ErrorCalculationESS();
            double result = ErrorTestingUtil.CalculateError(
                    calc,
                    ErrorTestingUtil.ACTUAL,
                    ErrorTestingUtil.IDEAL);
            Assert.AreEqual(1516.205, result, 0.001);
        }
    }
}
